serialisable.hpp

```diff
@@ -104,6 +108,23 @@ class Serialisable {
 		inline virtual double& getDouble() {
 			return value_;
 		}
+		inline void write(std::ostream& out, int = 0) {
+			out << value_;
+			double whole;
+			if (std::modf(value_, &whole) == 0)
+				out << ".0";
```

Comment:
The 'dot' tells apart `double` from `int`. In case a floating point value has no decimal value, the dot is added anyway to allow consistent deserialisation.